src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -109,138 +113,201 @@ public static boolean isWater(IBlockState iBlockState)
         {
-            item = Items.skull;
+            return Items.skull;
         }
-        else if(block instanceof BlockStem)
+        else if (blockState.getBlock() instanceof BlockStem)
         {
-            item =  block == Blocks.pumpkin ? Items.pumpkin_seeds
-                                                 : (block == Blocks.melon_block ? Items.melon_seeds
-                                                 : null);
+            return ((BlockStem) blockState.getBlock()).getSeedItem();
         }
-        else if(block instanceof BlockStoneSlab)
+        else if (blockState.getBlock() instanceof BlockStoneSlab)
         {
-            item = Item.getItemFromBlock(Blocks.stone_slab);
+            //Builder won't know how to build double stone slab
+            return Item.getItemFromBlock(Blocks.stone_slab);
         }
-        else if(block instanceof BlockStoneSlabNew)
+        else if (blockState.getBlock() instanceof BlockStoneSlabNew)
         {
-            item = Item.getItemFromBlock(Blocks.stone_slab2);
+            return Item.getItemFromBlock(Blocks.stone_slab2);
         }
-        else if(block instanceof BlockTripWire)
+        else if (blockState.getBlock() instanceof BlockTripWire)
         {
-            item = Items.string;
+            return Items.string;
         }
-        else if(block instanceof BlockWoodSlab)
+        else if (blockState.getBlock() instanceof BlockWoodSlab)
         {
-            item = Item.getItemFromBlock(Blocks.wooden_slab);
+            //Builder will also have trouble with double wood slab
+            return Item.getItemFromBlock(Blocks.wooden_slab);
         }
+        else
+        {
+            return Item.getItemFromBlock(blockState.getBlock());
+        }
+    }
 
-        return item;
+    private static int getDamageValue(Block block, IBlockState blockState)
+    {
+        if (block instanceof BlockCocoa)
+        {
+            return EnumDyeColor.BROWN.getDyeDamage();
+        }
+        else if (block instanceof BlockDirt)
+        {
+            return blockState.getValue(BlockDirt.VARIANT).getMetadata();
+        }
+        else if (block instanceof BlockDoublePlant
+                && blockState.getValue(BlockDoublePlant.HALF) == BlockDoublePlant.EnumBlockHalf.LOWER)
+        {
+            //If upper part we can't do much here
+            return blockState.getValue(BlockDoublePlant.VARIANT).getMeta();
+        }
+        else if (block instanceof BlockNewLeaf)
+        {
+            return block.getMetaFromState(blockState) & 3;
+        }
+        else if (block instanceof BlockOre)
+        {
+            return 0;
+        }
+        else if (block instanceof BlockSilverfish)
+        {
+            return block.getMetaFromState(blockState);
+        }
+        else if (block instanceof BlockSlab)
+        {
+            return block.damageDropped(blockState) & 7;
+        }
+        else if (block instanceof BlockTallGrass)
+        {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 297. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1871)